### PR TITLE
FSE: Add plans grid modal.

### PR DESCRIPTION
**Pull request by yansern**
#### Changes proposed in this Pull Request

This is a rough & dirty PR just makes plans grid display on block editor. All issues (styling/domains/data/etc) to be fixed in other PRs.

#### Testing instructions

After checking out this branch, run `yarn` once to install all the dependencies & recompile the `packages/plans-grid` package.

* Set `define( 'A8C_FSE_PLANS_GRID_ENABLE', true );` in `.config/sandbox.php` on your sandbox.
* Run `yarn dev --sync` on `apps\full-site-editing`.
* Go to `/wp-admin/post-new.php?post_type=page` on your sandbox.

#### Known issues

- Plans grid could not display price. To fix in another PR.

#### Screenshots

image

image

### CODE REVIEW

File: apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/index.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * File for various functionality which needs to be added to Simple and Atomic
+ * sites. The code in this file is always loaded in the block editor.
+ *
+ * Currently, this module may not be the best place if you need to load
+ * front-end assets, but you could always add a separate action for that.
```
Comment (simison) [inline apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/index.php:7]: The header text can be removed. :-)

File: apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/src/plans-modal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+const PlansModal: React.FunctionComponent< Props > = ( { isOpen, onClose, ...props } ) => {
+	if ( ! isOpen ) {
+		return null;
+	}
+
+	const header = (
+		<div>
+			<h1 className="wp-brand-font">{ __( 'Choose a plan', 'full-site-editing' ) }</h1>
```
Comment (a8ci18n) [inline apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/src/plans-modal/index.tsx:26]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **29** times:
`translate( 'Choose a plan' )` *ES Score: 7*

File: apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/src/plans-modal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+			<p>
+				{ __(
+					'Pick a plan that’s right for you. Switch plans as your needs change. There’s no risk, you can cancel for a full refund within 30 days.',
```
Comment (a8ci18n) [inline apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/src/plans-modal/index.tsx:29]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Pick a plan that’s right for you. Switch plans as your needs change. There’s no risk, you can cancel for a full refund within 30 days.' )` *ES Score: 11*

File: apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/index.php (hunk 1 above)
Comment (yansern) [reply apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/index.php]: Removed.

File: packages/plans-grid/src/base-styles.scss [hunk 4]
```diff
@@ -4,3 +4,33 @@
 @import '~@wordpress/base-styles/breakpoints';
 @import '~@wordpress/base-styles/mixins';
 @import '~@wordpress/base-styles/animations';
+
+// TODO: Typography needs to be shared. This is duplicated from `client/assets/stylesheets/shared/_typography.scss
```
Comment (simison) [inline packages/plans-grid/src/base-styles.scss:8]: Packaged these at https://github.com/Automattic/wp-calypso/pull/43718

File: apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/src/plans-modal/styles.scss [hunk 5]
```diff
@@ -0,0 +1,33 @@
+@import '~@automattic/typography/sass/fonts';
```
Comment (simison) [inline apps/full-site-editing/full-site-editing-plugin/editor-plans-grid/src/plans-modal/styles.scss:1]: Just FYI, if https://github.com/Automattic/wp-calypso/pull/43718 gets merged first, this line need rebasing.

**Comment by simison** [comment]
Re-running tests to see what's up. Rebasing can help too.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4128954

Thank you @yansern for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4128954).

**Pull request merged**